packages/cli/src/InternalHooks.ts

```diff
@@ -71,31 +76,69 @@ export class InternalHooks {
+
 		const info = {
-			version_cli: diagnosticInfo.versionCli,
-			db_type: diagnosticInfo.databaseType,
-			n8n_version_notifications_enabled: diagnosticInfo.notificationsEnabled,
-			n8n_disable_production_main_process: diagnosticInfo.disableProductionWebhooksOnMainProcess,
-			system_info: diagnosticInfo.systemInfo,
-			execution_variables: diagnosticInfo.executionVariables,
-			n8n_deployment_type: diagnosticInfo.deploymentType,
-			n8n_binary_data_mode: diagnosticInfo.binaryDataMode,
-			smtp_set_up: diagnosticInfo.smtp_set_up,
-			ldap_allowed: diagnosticInfo.ldap_allowed,
-			saml_enabled: diagnosticInfo.saml_enabled,
-			license_plan_name: diagnosticInfo.licensePlanName,
-			license_tenant_id: diagnosticInfo.licenseTenantId,
+			version_cli: N8N_VERSION,
+			db_type: config.getEnv('database.type'),
+			n8n_version_notifications_enabled: config.getEnv('versionNotifications.enabled'),
+			n8n_disable_production_main_process: config.getEnv(
+				'endpoints.disableProductionWebhooksOnMainProcess',
+			),
+			system_info: {
+				os: {
+					type: os.type(),
+					version: os.version(),
+				},
+				memory: os.totalmem() / 1024,
+				cpus: {
+					count: cpus.length,
+					model: cpus[0].model,
+					speed: cpus[0].speed,
+				},
+			},
+			execution_variables: {
+				executions_mode: config.getEnv('executions.mode'),
+				executions_timeout: config.getEnv('executions.timeout'),
+				executions_timeout_max: config.getEnv('executions.maxTimeout'),
+				executions_data_save_on_error: config.getEnv('executions.saveDataOnError'),
+				executions_data_save_on_success: config.getEnv('executions.saveDataOnSuccess'),
+				executions_data_save_on_progress: config.getEnv('executions.saveExecutionProgress'),
+				executions_data_save_manual_executions: config.getEnv(
+					'executions.saveDataManualExecutions',
+				),
+				executions_data_prune: config.getEnv('executions.pruneData'),
+				executions_data_max_age: config.getEnv('executions.pruneDataMaxAge'),
+			},
+			n8n_deployment_type: config.getEnv('deployment.type'),
+			n8n_binary_data_mode: binaryDataConfig.mode,
+			smtp_set_up: config.getEnv('userManagement.emails.mode') === 'smtp',
+			ldap_allowed: authenticationMethod === 'ldap',
+			saml_enabled: authenticationMethod === 'saml',
+			license_plan_name: this.license.getPlanName(),
+			license_tenant_id: config.getEnv('license.tenantId'),
+			binary_data_s3: isS3Available && isS3Selected && isS3Licensed,
+			multi_main_setup_enabled: config.getEnv('multiMainSetup.enabled'),
 		};
 
+		const firstWorkflow = await this.workflowRepository.findOne({
+			select: ['createdAt'],
+			order: { createdAt: 'ASC' },
+			where: {},
+		});
+
 		return await Promise.all([
 			this.telemetry.identify(info),
 			this.telemetry.track('Instance started', {
 				...info,
-				earliest_workflow_created: earliestWorkflowCreatedAt,
+				earliest_workflow_created: firstWorkflow?.createdAt,
```

Comment:
We could track creations in workflow statistics but we don't, so this is only the earliest surviving one. Maybe this should be renamed `earliest_undeleted_workflow_created`?

Reply:
Yeah. but `earliest_undeleted_workflow_created` sounds too verbose, and also unlikely the data-point that we added this code for. We definitely need a better mechanism to know the first workflow that was created, irrespective of it's deleted.